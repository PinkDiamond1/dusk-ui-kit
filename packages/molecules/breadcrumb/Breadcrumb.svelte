<script>
  import { createEventDispatcher, getContext } from "svelte";
  import contexts from "@dusk-network/helpers/contexts.js";
  import Icon from "@dusk-network/icon/Icon.svelte";
  import Item from "@dusk-network/breadcrumb/Item.svelte";
  import "./styles.css";

  export let href;

  const context = getContext("DUK:breadcrumb:context");
  const dispatch = createEventDispatcher();
  const handleClick = () => {
    dispatch("exit");
  };
</script>

<ul
  class="{$$props.class || ''} duk-breadcrumb"
  class:duk-breadcrumb--wizard="{context === contexts.BREADCRUMB.WIZARD}"
>
  <Item href="{href}" on:click="{handleClick}">
    <Icon name="backspace-outline" />
  </Item>
  <slot />
</ul>
