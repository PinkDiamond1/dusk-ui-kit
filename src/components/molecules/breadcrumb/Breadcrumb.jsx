/* @flow */
/** @jsx jsx */
import { Global, jsx } from '@emotion/core'
import * as React from 'react'
import BootstrapBreadcrumb from 'react-bootstrap/Breadcrumb'

import BreadcrumbItem from '../../atoms/breadcrumb-item/BreadcrumbItem'
import theme from '../../../config/dusk.theme'

type Props = {
  children: React.Node,
  className: string,
}

const Breadcrumb = (props: Props): React.Element<*> => {
  const { children, className } = props
  const updatedClassName = `__duk-breadcrumb ${className}`
  return (
    <>
      <Global
        styles={{
          '.__duk-breadcrumb .breadcrumb-item': {
            a: {
              opacity: 0.5,
              color: `${theme.colors.gray900}`,
            },
            '&:last-child': {
              a: {
                opacity: 1,
              },
            },
          },
        }}
      />
      <BootstrapBreadcrumb className={updatedClassName}>{children}</BootstrapBreadcrumb>
    </>
  )
}

Breadcrumb.defaultProps = {
  className: '',
}

Breadcrumb.Item = BreadcrumbItem

export default Breadcrumb
