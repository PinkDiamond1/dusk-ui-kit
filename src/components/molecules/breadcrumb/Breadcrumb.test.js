import React from 'react'
import renderer from 'react-test-renderer'

import Breadcrumb from './Breadcrumb'

it('test to see if the Breadcrumb renders correctly', () => {
  const tree = renderer.create(<Breadcrumb>Smoke Test</Breadcrumb>).toJSON()
  expect(tree).toMatchSnapshot()
})
