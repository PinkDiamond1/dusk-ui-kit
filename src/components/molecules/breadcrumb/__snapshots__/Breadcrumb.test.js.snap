// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Breadcrumb renders correctly 1`] = `
<nav
  aria-label="breadcrumb"
  className="__duk-breadcrumb "
>
  <ol
    className="breadcrumb"
  >
    Smoke Test
  </ol>
</nav>
`;
